### PR TITLE
Enable ntfs-3g support in ntfsprogs

### PR DESCRIPTION
**Pull request by xen0l**

### CODE REVIEW

**Comment by alarcher** [comment]
Is there a difference between the Tuxera tarball and the work by JP André http://jp-andre.pagesperso-orange.fr/openindiana-ntfs-3g.html ?

**Comment by xen0l** [comment]
No idea, according to JPA's website, there should be. Do you use it?

**Comment by alarcher** [comment]
Since he worked on fixing issues for OI I just thought that using his version would be best.
I used some 2015 version but not extensively.
Maybe worth talking to him.

File: components/sysutils/ntfsprogs/ntfsprogs.p5m [hunk 1]
```diff
@@ -65,19 +73,28 @@ file path=usr/include/ntfs-3g/types.h
 file path=usr/include/ntfs-3g/unistr.h
 file path=usr/include/ntfs-3g/volume.h
 file path=usr/include/ntfs-3g/xattrs.h
+file path=usr/lib/libntfs-3g.a
+file path=usr/lib/libntfs-3g.la
```
Comment (pyhalov) [inline components/sysutils/ntfsprogs/ntfsprogs.p5m]: avoid mentioning *.la files

File: components/sysutils/ntfsprogs/ntfsprogs.p5m [hunk 2]
```diff
@@ -65,19 +73,28 @@ file path=usr/include/ntfs-3g/types.h
...
 link path=usr/lib/libntfs-3g.so target=libntfs-3g.so.87.0.0
 link path=usr/lib/libntfs-3g.so.87 target=libntfs-3g.so.87.0.0
 file path=usr/lib/libntfs-3g.so.87.0.0
 file path=usr/lib/pkgconfig/libntfs-3g.pc
+file files/fstyp path=usr/lib/fs/ntfs-3g/fstyp
```
Comment (pyhalov) [inline components/sysutils/ntfsprogs/ntfsprogs.p5m:80]: should it have permissions explicitly set?

File: components/sysutils/ntfsprogs/Makefile [hunk 3]
```diff
@@ -11,12 +11,14 @@
...
 #
 # Copyright 2014 Alexander Pyhalov.  All rights reserved.
+# Copyright 2016 Adam Stevko.  All rights reserved.
 #
 
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME= ntfs-3g_ntfsprogs
 COMPONENT_VERSION= 2016.2.22
+BUILD_VERSION=	   1
```
Comment (pyhalov) [inline components/sysutils/ntfsprogs/Makefile]: COMPONENT_REVISION

File: components/sysutils/ntfsprogs/ntfsprogs.p5m (hunk 2 above)
Comment (xen0l) [inline components/sysutils/ntfsprogs/ntfsprogs.p5m:80]: I have a transform for this

**Comment by jpandre** [comment]
The differences between the latest Tuxera release of ntfs-3g (2016.2.22) and mine (2016.2.22AR.1) are the ones dated Apr 1st on http://jp-andre.pagesperso-orange.fr/changelog.html
In particular, the ability to execute a plugin for reading the new compressed file format used by Windows 10. Both have support for OpenIndiana.

**Comment by jpandre** [comment]
I see you have copied an issue from the Linux version of fuse, but the equivalent issue in the kernel module for OpenIndiana has been fixed (http://jp-andre.pagesperso-orange.fr/openindiana-ntfs-3g.html or https://github.com/jurikm/illumos-fusefs/commits/master/ see the commits for june 2015)

> libfuse/libfuse Without `default_permissions`, cached permissions are only checked on first access bug needs-kernel-support
> #15 opened on Jan 30 by jpandre

**Comment by xen0l** [comment]
@jpandre I based fuse package on sources from https://github.com/jurikm/illumos-fusefs/commits/master/. ntfsprogs is something, which needs to be updated for your sources.

**Comment by jpandre** [comment]
Hi Adam,

Adam Števko wrote:

> @jpandre https://github.com/jpandre I based fuse package on sources
> from https://github.com/jurikm/illumos-fusefs/commits/master/. ntfsprogs
> is something, which needs to be updated for your sources.

Note : ntfsprogs (the tools) are unrelated to fuse and libfuse.
Only the drivers (ntfs-3g and lowntfs-3g) use them. The shared
code between the drivers and the tools are in libntfs-3g

The sources for fuse (the kernel module) are the same on
github and on my website. On github I have promoted it as
version 1.3 but there is no difference from 1.2AR.7.

Regards

Jean-Pierre

**Comment by pyhalov** [comment]
there are too much commits here...

**Comment by jpandre** [comment]
Hi Alexander

Alexander Pyhalov wrote:

> there are too much commits here...

???

**Comment by pyhalov** [comment]
First commit doesn't belong to this pull request

**Comment by xen0l** [comment]
Fixed, I'll squash commits. Could this get another round of testing?

**Comment by jpandre** [comment]
Adam Števko wrote:

> Fixed, I'll squash commits. Could this get another round of testing?
> 
> I have no personal experience with external fuse (--with-fuse=external),
> has this configuration been tested by somebody ?

Jean-Pierre

**Comment by xen0l** [comment]
I did few tests like creating ntfs on zvol, mounting it etc. It worked.

Adam

File: components/sysutils/ntfsprogs/ntfsprogs.p5m [hunk 4]
```diff
@@ -10,8 +10,11 @@
 
 #
 # Copyright 2014 Alexander Pyhalov. All rights reserved.
+# Copyright 2016 Adam Stevko. All rights reserved.
 #
 
+<transform file path=usr/lib/fs/fuse/(.*) -> default mode 0555>
```
Comment (pyhalov) [inline components/sysutils/ntfsprogs/ntfsprogs.p5m]: is pattern correct?

**Pull request merged**